File: task-03/C/prime_numbers_upto_n.c
#include <stdio.h>

int main() {
    int n, i, j, isPrime;

    printf("Enter a number: ");
    scanf("%d", &n);

    printf("Prime numbers up to %d are:\n", n);

    for(i = 2; i <= n; ++i) {
        isPrime = 1;

        for(j = 2; j <= i/2; ++j) {
            if(i % j == 0) {
                isPrime = 0;
                break;
            }
        }

        if(isPrime) {
            printf("%d ", i);
        }
    }

    return 0;
}

File: task-03/Ruby/prime_numbers_upto_n.rb
puts "Enter a number:"
n = gets.chomp.to_i

prime_numbers = []

for i in 2..n
 is_prime = true

 for j in 2..Math.sqrt(i).to_i
    if i % j == 0
      is_prime = false
      break
    end
 end

 prime_numbers.push(i) if is_prime
end

puts "Prime numbers up to #{n} are:"
puts prime_numbers